sklearn/manifold/spectral_embedding_.py

```diff
@@ -4,508 +4,26 @@
-        if self.affinity == 'rbf':
-            self.gamma_ = (self.gamma
-                           if self.gamma is not None else 1.0 / X.shape[1])
-            self.affinity_matrix_ = rbf_kernel(X, gamma=self.gamma_)
-            return self.affinity_matrix_
-        self.affinity_matrix_ = self.affinity(X)
-        return self.affinity_matrix_
-
-    def fit(self, X, y=None):
-        """Fit the model from data in X.
-
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_features)
-            Training vector, where n_samples is the number of samples
-            and n_features is the number of features.
-
-            If affinity is "precomputed"
-            X : array-like, shape (n_samples, n_samples),
-            Interpret X as precomputed adjacency graph computed from
-            samples.
-
-        Returns
-        -------
-        self : object
-            Returns the instance itself.
-        """
-
-        X = check_array(X, ensure_min_samples=2, estimator=self)
-
-        random_state = check_random_state(self.random_state)
-        if isinstance(self.affinity, six.string_types):
-            if self.affinity not in set(("nearest_neighbors", "rbf",
-                                         "precomputed")):
-                raise ValueError(("%s is not a valid affinity. Expected "
-                                  "'precomputed', 'rbf', 'nearest_neighbors' "
-                                  "or a callable.") % self.affinity)
-        elif not callable(self.affinity):
-            raise ValueError(("'affinity' is expected to be an an affinity "
-                              "name or a callable. Got: %s") % self.affinity)
-
-        affinity_matrix = self._get_affinity_matrix(X)
-        self.embedding_ = spectral_embedding(affinity_matrix,
-                                             n_components=self.n_components,
-                                             eigen_solver=self.eigen_solver,
-                                             random_state=random_state)
-        return self
-
-    def fit_transform(self, X, y=None):
-        """Fit the model from data in X and transform X.
+    return laplacian_eigenmap(adjacency, n_components=n_components,
+                              eigen_solver=eigen_solver, drop_first=drop_first,
+                              random_state=random_state, eigen_tol=eigen_tol,
+                              norm_laplacian=norm_laplacian)
+spectral_embedding.__doc__ = laplacian_eigenmap.__doc__
 
-        Parameters
-        ----------
-        X: array-like, shape (n_samples, n_features)
-            Training vector, where n_samples is the number of samples
-            and n_features is the number of features.
 
-            If affinity is "precomputed"
-            X : array-like, shape (n_samples, n_samples),
-            Interpret X as precomputed adjacency graph computed from
-            samples.
+class SpectralEmbedding(LaplacianEigenmap):
```

Comment:
@amueller @GaelVaroquaux We need to subclass here to avoid sideeffects propagating to the original class LaplacianEigenmap (as deprecated modifies the `__init__` of the passed class object)